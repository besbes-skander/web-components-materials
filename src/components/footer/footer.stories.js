import './footer'

export default {
    title: 'Components/Footer',
}

export const Default = () => '<arl-footer></arl-footer>'
export const FooterWithConnectedUser = () => '<arl-footer is-user-connected="true"></arl-footer>'
export const FooterWithDisconnectedUser = () => '<arl-footer is-user-connected="false"></arl-footer>'