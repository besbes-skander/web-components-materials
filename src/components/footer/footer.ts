import { CustomHTMLElement } from '../../utils'

const template = document.createElement('template')

function createTemplate(text: string): string {
    return `
        <div id="footer">
            <h3>Besoin d'aide ?</h3>
            <p>${text}</p>
        </div>
    `
}

export class Footer extends CustomHTMLElement {
    constructor() {
        super()

        this.attachShadow({ mode: 'open' })
            .appendChild(template.content.cloneNode(true))

        this.render()
    }

    static get observedAttributes() {
        return ['is-user-connected']
    }

    attributeChangedCallback(attributeName: string, _oldValue: string, newValue: string) {
        if (attributeName !== 'is-user-connected') {
            return
        }

        this.render(newValue)
    }

    render(isUserConnected?: string) {
        const footerText = (isUserConnected === 'true')
            ? 'Contact | Plan | Deconnexion'
            : 'Contact | Plan | Connexion'

        const newTemplate = createTemplate(footerText)
        this.renderComponent(newTemplate)
    }
}

customElements.define('arl-footer', Footer)