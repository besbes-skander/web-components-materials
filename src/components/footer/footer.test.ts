import { render } from '../../utils'
import { Footer } from './footer'

it('should return footer when user is connected', async () => {
    const footer = await render(Footer)
    footer.setAttribute('is-user-connected', 'true')
    expect(footer.shadowRoot?.querySelector('#footer')).toMatchSnapshot()
})

it('should return footer when user is NOT connected', async () => {
    const footer = await render(Footer)
    footer.setAttribute('is-user-connected', 'false')
    expect(footer?.shadowRoot?.querySelector('#footer')).toMatchSnapshot()
})