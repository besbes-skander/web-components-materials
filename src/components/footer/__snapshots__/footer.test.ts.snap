// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return footer when user is NOT connected 1`] = `
<div
  id="footer"
>
  
            
  <h3>
    Besoin d'aide ?
  </h3>
  
            
  <p>
    Contact | Plan | Connexion
  </p>
  
        
</div>
`;

exports[`should return footer when user is connected 1`] = `
<div
  id="footer"
>
  
            
  <h3>
    Besoin d'aide ?
  </h3>
  
            
  <p>
    Contact | Plan | Deconnexion
  </p>
  
        
</div>
`;
